### DISCUSSION TITLE
How does VPS work with the `vite` command?

### DISCUSSION THREAD

**Discussion by AaronBeaudoin** [vikejs/vike#827] [Discussion]
It used to be the case that we needed some `server.js` file to use VPS, but more recently VPS has been able to work by simply running the `vite` command directly. How does it do that?

**Comment by brillout** [comment]
See `addSsrMiddleware()` in VPS's source code.

**Comment by JadynCarroll** [comment]
Hey there! I'm new to this forum, but I can help you out with your question. So, when it comes to using VPS with the vite command, things have changed a bit recently. In the past, you needed a server.js file to use VPS, but now you can just run the vite command directly. The reason for this change is that [ishosting vps](https://ishosting.com/en/vps) has been updated to be more user-friendly and make things easier for developers. Essentially, vps is hosting the vite command for you, so you don't need to worry about setting up a server.js file or anything like that. It's all taken care of behind the scenes. It simplifies the process and makes things much easier for developers. Hope that helps!

**Reply by brillout** [reply]
Thanks for the answer! FYI I've PM'd with OP and Aaron is interested in how Vite(-plugin-ssr) is able to automagically add itself to Vite's dev server.

**Comment by alexblackkot9** [comment]
Hey there! I'm new to this forum, but I can help you out with your question. So, when it comes to using VPS with the vite command, things have changed a bit recently. In the past, you needed a server.js file to use VPS, but now you can just run the vite command directly.

**Comment by JadynCarroll** [comment]
Hey there! I'm new to this forum, but I can help you out with your question. So, when it comes to using VPS with the vite command, things have changed a bit recently. In the past, you needed a server.js file to use VPS, but now you can just run the vite command directly. The reason for this change is that [ishosting vps](https://ishosting.com/en/vps) has been updated to be more user-friendly and make things easier for developers. Essentially, vps is hosting the vite command for you, so you don't need to worry about setting up a server.js file or anything like that. It's all taken care of behind the scenes. It simplifies the process and makes things much easier for developers. Hope that helps!
